documentation/new-version-guideline.md

```diff
@@ -0,0 +1,342 @@
+
+# Azure Go Management SDK Guideline
+
+Azure Go management SDK follows the [new Azure SDK guidelines](https://azure.github.io/azure-sdk/general_introduction.html), try to create easy-to-use APIs that are idiomatic, compatible, and dependable.
+
+You can find the full list of management modules [here](https://pkg.go.dev/github.com/Azure/azure-sdk-for-go/sdk).
+
+In this guideline, we will give some instructions about the API usage pattern as well as trouble shooting method. For those are new to management Go SDK, please refer to [quickstart](./new-version-quickstart.md). For those migrate from older versions of management Go SDK, please refer to [migration guide](https://aka.ms/azsdk/go/mgmt/migration).
+
+## Pageable Operations
+
+### General usage
+
+Pageable operations return final data over multiple GET requests. Each GET will receive a page of data consisting of a slice of items. You need to use New*Pager to create a pager helper for all pageable operations. With the returned `*runtime.Pager[T]`, you can fetch pages and determine if there are more pages to fetch. For examples:
+
+```go
+import "github.com/Azure/azure-sdk-for-go/sdk/resourcemanager/resources/armresources"
+```
+
+```go
+ctx := context.TODO() // your context
+pager := rgClient.NewListPager(nil)
+var resourceGroups []*armresources.ResourceGroup
+for pager.More() {
+    nextResult, err := pager.NextPage(ctx)
+    if err != nil {
+        // handle error...
+    }
+    if nextResult.ResourceGroupListResult.Value != nil {
+        resourceGroups = append(resourceGroups, nextResult.ResourceGroupListResult.Value...)
+    }
+}
+// dealing with `resourceGroups`
+```
+
+> NOTE: No IO calls are made until the NextPage() method is invoked. The read consistency across pages is determined by the service implement.
+
+### Item iterator
+
+If you do not care about the underlaying detail about the pageable operation, you can use the following generic utility to create a per-item iterator for all pageable operation.
+
+***Item iterator utility***
+
+```go
+import "github.com/Azure/azure-sdk-for-go/sdk/azcore/runtime"
+```
+
+```go
+type PageConstraint[TItem any] interface {
+	Items() []*TItem
+}
+
+type Iterator[TItem any, TPage PageConstraint[TItem]] struct {
```

Comment:
Note that customers can't implement this at present due to the missing `Items()` method on the response envelopes for paged responses.  My understanding is that we'd omit this for now until customers ask for it.

CC @JeffreyRichter